src/Scripting/CSharpTest/ScriptTests.cs

```diff
@@ -951,6 +951,53 @@ public void CreateScriptWithNullableContextWithCSharp8()
                 Diagnostic(ErrorCode.WRN_NullAsNonNullable, "null").WithLocation(2, 28)
             );
         }
+        
+        [Fact, WorkItem(49529, "https://github.com/dotnet/roslyn/issues/49529")]
+        public async Task SwitchPatternWithVar_WhenValidCode_ShouldReturnValidResult()
+        {
+            var code = @"
+using System;
+var data = ""data"";
+var reply = data switch {
+    null => ""data are null"",
+    """" => ""data are empty"",
+    _ => data
+};
+
+return reply;
+";
+            var script = CSharpScript.Create(code, ScriptOptions.Default.WithLanguageVersion(LanguageVersion.CSharp8));
+            var compilation = script.GetCompilation();
+            compilation.VerifyDiagnostics();
+            
+            var result = await script.EvaluateAsync();
+            
+            Assert.Equal("data", result);
+            
+        }
+        
+        [WorkItem(49529, "https://github.com/dotnet/roslyn/issues/49529")]
+        [Fact]
+        public async Task SwitchPatternWithVar_WhenInvalidCode_ShouldReturnTheCompilationErrors()
```

Comment:
Consider a test where the pattern has an error, and a test where the arm has an error.